src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerRejectedFT.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.divorce.caseworker;
+
+import io.restassured.response.Response;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.hmcts.divorce.caseworker.event.CaseworkerRejected.CASEWORKER_REJECTED;
+import static uk.gov.hmcts.divorce.divorcecase.NoFaultDivorce.CASE_TYPE;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.Submitted;
+import static uk.gov.hmcts.divorce.testutil.CaseDataUtil.caseData;
+import static uk.gov.hmcts.divorce.testutil.TestConstants.ABOUT_TO_SUBMIT_URL;
+
+
+@SpringBootTest
+public class CaseworkerRejectedFT extends FunctionalTestSuite {
+
+    private static final String REQUEST = "classpath:request/casedata/ccd-callback-caseworker-reject-application-about-to-submit.json";
+
+    @Test
+    public void shouldSetPreviousStateWhenAboutToSubmitCallbackIsSuccessful() throws Exception {
+
+        // This caseId will need to be updated if the cases in AAT are cleared
```

Comment:
Happy to be totally wrong here but should this not be a new case each time? (Can be cleared up at the end of the tests if necessary) Also bear in mind these tests will be running on Preview, demo, ITHC & perftest - so can't rely on previously created cases

Reply:
There's still rather a lot of clean up required, that test doesn't look like something we'd have actually written. With this PR we can create a new case and can therefore start build our own workflows. I've had to add a new ticket to manage the clean-up due to holidays and absences and so that we don't add new NFDiv business logic to this PR.